File: sort/bubbleSort.js
const bubbleSort = (arr) =>{
    const result = [...arr];
    let tempInt;
    
    for (let i = result.length; i > 0; i--) {
        for (let j = 1; j < i; j++) {
            if(result[j-1] > result[j]){
                tempInt = result[j];
                result[j] = result[j-1];
                result[j-1] = tempInt;
            }
        }
    }
    return result
}

let arr = [9, 4, 3, 10, 5, 8, 7, 6, 2, 1];


console.log(arr);

arr = bubbleSort(arr);

console.log(arr);

File: sort/bubbleSort.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> bubbleSort(vector<int> arr){
    int temp;

    for(int i = arr.size(); i > 0; i--){
        for(int j = 1; j < i ; j++){
            if(arr[j-1] > arr[j]){
                temp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = temp;
            }
        }
    }
    return arr;
}

int main(){

    vector<int> arr = {9, 4, 3, 10, 5, 8, 7, 6, 2, 1};


    for(int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }
    cout << endl;
    
    arr = bubbleSort(arr);

    for(int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }
    cout << endl;









    return 0;
}